packages/shylu/shylu_node/basker/src/shylubasker_solve_rhs.hpp

```diff
@@ -582,22 +604,39 @@ namespace BaskerNS
       #ifdef BASKER_DEBUG_SOLVE_RHS
       printf("Lower Solve blk (%d, 0): size=(%dx%d) srow=%d, scol=%d\n",b,(int)L.nrow,(int)L.ncol, (int)L.srow,(int)L.scol);
       printf("[\n");
-      for (Int i = 0; i < gn; i++) printf( "%d  %e %e\n",i,x(i),y(i));
+      for (Int i = 0; i < gn; i++) printf( "%d %.16e %.16e\n",i,x(i),y(i));
       printf( "];\n");
+      if (b == 3) {
+        const Int bcol = L.scol + scol_top;
+        const Int brow = L.scol + scol_top;
+        printf( " P = [\n" );
+        for (Int k = 0; k < L.ncol; k++) printf( "%d %d\n",brow+k,gperm(brow+k) );
+        printf("];\n");
+        printf( " L = [\n" );
+        for(Int k = 0; k < L.ncol; ++k) {
+          for (Int i = L.col_ptr(k); i < L.col_ptr(k+1); i++) {
+            printf( "%d %d %e\n",L.row_idx(i),k,L.val(i) );
+          }
+        }
+        printf("];\n");
+      }
       #endif
 
       //Update offdiag
       for(Int bb = 1; bb < LL_size(b); ++bb)
       {
         BASKER_MATRIX &LD = LL(b)(bb);
         //x = LD*y;
+        #ifdef BASKER_DEBUG_SOLVE_RHS
```

Comment:
All the print output is commented out, should enable some of this if guarded by the debug macro